homeassistant/components/plugwise/climate.py

```diff
@@ -122,49 +144,77 @@ def supported_features(self):
     def device_state_attributes(self):
         """Return the device specific state attributes."""
         attributes = {}
-        attributes["outdoor_temperature"] = self._outdoor_temperature
-        attributes["available_schemas"] = self._api.get_schema_names(
-            self._domain_objects
-        )
-        attributes["active_schema"] = self._active_schema
+        if self._outdoor_temperature:
+            attributes["outdoor_temperature"] = self._outdoor_temperature
+        attributes["available_schemas"] = self._schema_names
+        attributes["selected_schema"] = self._selected_schema
+        if self._illuminance is not None:
+            attributes["illuminance"] = self._illuminance
+        if self._boiler_temperature:
+            attributes["boiler_temperature"] = self._boiler_temperature
+        if self._water_pressure:
+            attributes["water_pressure"] = self._water_pressure
         return attributes
 
-    def update(self):
-        """Update the data from the thermostat."""
-        _LOGGER.debug("Update called")
-        self._domain_objects = self._api.get_domain_objects()
-        self._outdoor_temperature = self._api.get_outdoor_temperature(
-            self._domain_objects
-        )
-        self._active_schema = self._api.get_active_schema_name(self._domain_objects)
+    @property
+    def preset_modes(self):
+        """Return the available preset modes list and make the presets with their temperatures available."""
+        return self._presets_list
+
+    @property
+    def hvac_modes(self):
+        """Return the available hvac modes list."""
+        if self._heating_status is not None:
+            if self._cooling_status is not None:
+                return self._hvac_modes_2
+            return self._hvac_modes_1
 
```

Comment:
Please add an explicit `return None` at the end. We need to return something explicitly for all cases if we return a value anywhere in a function.

Reply:
Updated in several places where a `return None` was missing.